bee-storage/bee-storage-rocksdb/src/access/batch.rs

```diff
@@ -39,8 +40,13 @@ impl BatchBuilder for Storage {
         let mut write_options = WriteOptions::default();
         write_options.set_sync(false);
         write_options.disable_wal(!durability);
+
+        let guard = batch.should_lock.then(|| self.locks.message_id_to_metadata.read());
```

Comment:
If we had more lock we would need to lock them all because we can't know ahead what CFs will be used ?